File: Chapter16/SampleViewAnimation/Java/SampleViewAnimation/app/src/main/java/com/ai/android/ExerciseSystemIntents/MainActivity.java
package com.ai.android.ExerciseSystemIntents;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        super.onCreateOptionsMenu(menu);

        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.main_menu,menu);

        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        handleMenus(item);
        return true;
    }

    private void handleMenus(MenuItem item) {
        appendMenuItemText(item);
        switch (item.getItemId()) {
            case R.id.menu_clear:
                emptyText();
                break;

            case R.id.menu_list_animation:
                Intent intent = new Intent(this, ViewAnimationActivity.class);
                startActivity(intent);
                break;
        }
    }

    private void appendMenuItemText(MenuItem item) {
        String title = item.getTitle().toString();
        TextView tv = findViewById(R.id.textViewId);
        tv.setText(String.format("%s\n%s:%d", tv.getText(), title, item.getItemId()));
    }

    private void emptyText() {
        TextView tv = findViewById(R.id.textViewId);
        tv.setText("");
    }
}

File: Chapter16/SampleViewAnimation/Kotlin/SampleViewAnimation/app/src/main/java/com/ai/android/sampleviewanimation/MainActivity.kt
package com.ai.android.sampleviewanimation

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        super.onCreateOptionsMenu(menu)

        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        handleMenus(item)
        return true
    }

    private fun handleMenus(item: MenuItem) {
        appendMenuItemText(item)
        when (item.itemId) {
            R.id.menu_clear -> emptyText()
            R.id.menu_list_animation -> {
                val intent = Intent(this, ViewAnimationActivity::class.java)
                startActivity(intent)
            }
        }
    }

    private fun appendMenuItemText(item: MenuItem) {
        val tv = findViewById<TextView>(R.id.textViewId)
        tv.text = String.format("%s\n%s:%d", tv.text, item.title, item.itemId)
    }

    private fun emptyText() {
        val tv = findViewById<TextView>(R.id.textViewId)
        tv.text = ""
    }

    companion object {
        const val TAG = "MainActivity"
    }
}